pkg/cmd/server/start/start_master.go

```diff
@@ -429,7 +429,7 @@ func StartAPI(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) error {
 	}
 
 	// verify we can connect to etcd with the provided config
-	if err := etcd.TestEtcdClient(oc.EtcdClient); err != nil {
+	if _, err := etcd.GetAndTestEtcdClient(oc.Options.EtcdClientInfo); err != nil {
```

Comment:
defer closing of the client which will turn off keepalive? https://github.com/openshift/origin/blob/2c59a8b0fa484139d7079f5bb39f37d360ee11cf/Godeps/_workspace/src/github.com/coreos/go-etcd/etcd/client.go#L187-187

That said, if it still makes sense to keep the etcdClient on the master config I'm not arguing against it.  The goal of the bug was to see if it was still being passed through to storage (looks like it isn't) and investigate if we still needed it on the config.  If reusing the client makes sense then let's reuse it.